notion-server/src/ethereum.rs

```diff
@@ -1,4 +1,77 @@
+use base64;
 use std::collections::HashMap;
+use std::convert::TryFrom;
+use snafu::{Snafu, ResultExt, Backtrace};
+
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    Decode {
+        backtrace: Backtrace,
+        source: base64::DecodeError,
+    },
+    OutOfBounds {
+        message: String,
+    }
+}
+
+pub mod interface_args {
```

Comment:
Why the rename?

Reply:
`args` felt too generic, and it make me uncomfortable to have eg. `args::ShardBlock` and `ShardBlock`, because it wasn't immediately apparent from the module name and struct names what the difference once.  IMO `interface_args` makes it MUCH more immediately clear what the differences between the structs are, and will make 1) it easier for us to talk about these 2 structs and quickly distinguish (eg. the public-facing interface args for `interface_args::ShardBlock` vs the internal `ShardBlock`) and 2) seems like this would also be a more immediately-obvious distinction if people new to this codebase are reading it for the first time.

Interested in your opinions here also and looking forward to discussing, but that was my logic.

Reply:
`notion_server::ethereum::ShardBlock` will never be visible outside of the `ethereum` module, so outside users won't be confused. They won't even be aware there are two similarly named structs.

The only code which has to refer to both `args::ShardBlock` and `ShardBlock` lives under the `ethereum` module, so I think that'll be plenty clear.

If it gets particularly confusing, we can always rename the module as we go.